### PR TITLE
REF: ujson cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/include/pandas/vendored/ujson/lib/ultrajson.h [hunk 1]
```diff
@@ -137,6 +137,9 @@ typedef int64_t JSLONG;
 #error "Endianness not supported"
 #endif
 
+// PANDAS_FORMAT is *not* in the original file this is vendored from
```
Comment (WillAyd) [inline pandas/_libs/include/pandas/vendored/ujson/lib/ultrajson.h:140]: Any reason to move this? Unless it is used elsewhere should just stay in the module

Comment (jbrockmendel) [reply pandas/_libs/include/pandas/vendored/ujson/lib/ultrajson.h:140]: i think i did this for a branch where i was trying to reimplement bits of the objToJSON.c in cython.  will revert this bit.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Cool nice work - merge on green

**Review by mroeschke — APPROVED** [review]

**Comment by neutrinoceros** [comment]
This breaks user code that relied on `pandas.io.json.loads`, I'm surprised this went in without a deprecation period or a mention in pandas 2.2's changelog. Is this not considered public API ? (the lack of any trailing underscore suggests it's not private)

**Comment by mroeschke** [comment]
`pandas.io.json.loads` is not a public API. The public JSON methods are listed in https://pandas.pydata.org/docs/reference/io.html#json and the more general policy on what's public is stated in https://pandas.pydata.org/docs/reference/index.html#api-reference

**Pull request merged**